File: Módulo Python/Códigos/4 - Estrutura de Repetição/Exercicio.11.py
#11. Faça um algoritmo para ler 100 números, calcular a soma
# dos números, a média e o maior e menor número encontrados. 

soma = 0.0
maior = float('-inf')
menor = float('inf')

for i in range(100):
    numero = float(input(f"Digite o {i+1}º número: "))
    soma += numero
    if numero > maior:
        maior = numero
    if numero < menor:
        menor = numero

media = soma / 100
print(f"Soma dos números: {soma}")
print(f"Média dos números: {media}")
print(f"Maior número: {maior}")
print(f"Menor número: {menor}")

File: Livro - Módulo Java/4 - Estrutura de Repetição/Exercicio11.java
import java.util.Scanner;

public class Exercicio11 {
    public static void main(String[] args) {
        // Faça um algoritmo para ler 100 números, calcular a soma dos números, a média e o maior e menor número encontrados

        Scanner scanner = new Scanner(System.in);
        double[] numeros = new double[100];

        for (int i = 0; i < 100; i++) {
            System.out.print("Digite o " + (i + 1) + "º número: ");
            numeros[i] = scanner.nextDouble();
        }

        calcularEExibirEstatisticas(numeros);
    }

    public static void calcularEExibirEstatisticas(double[] numeros) {
        double soma = 0;
        double maior = Double.NEGATIVE_INFINITY;
        double menor = Double.POSITIVE_INFINITY;

        for (int i = 0; i < numeros.length; i++) {
            double numero = numeros[i];

            soma += numero;

            if (numero > maior) {
                maior = numero;
            }

            if (numero < menor) {
                menor = numero;
            }
        }

        double media = soma / numeros.length;

        System.out.println("Resultados");
        System.out.println("Soma: " + soma);
        System.out.println("Média: " + media);
        System.out.println("Maior Número: " + maior);
        System.out.println("Menor Número: " + menor);
    }
}